java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -54,14 +54,17 @@ public class ColumnView implements AutoCloseable, BinaryOperable {
     this.rows = ColumnView.getNativeRowCount(viewHandle);
     this.nullCount = ColumnView.getNativeNullCount(viewHandle);
     this.offHeap = null;
+    boolean success = false;
     try {
       AssertEmptyNulls.assertNullsAreEmpty(this);
-    } catch (AssertionError ae) {
-      // offHeap state is null, so there is nothing to clean in offHeap
-      // delete ColumnView to avoid memory leak
-      deleteColumnView(viewHandle);
-      viewHandle = 0;
-      throw ae;
+      success = true;
+    } finally {
+      if (!success) {
+        // offHeap state is null, so there is nothing to clean in offHeap
+        // delete ColumnView to avoid memory leak
+        deleteColumnView(viewHandle);
+        viewHandle = 0;
+      }
```

Comment:
Using a boolean seems unnecessary here, we simply need to cleanup if anything is thrown.  For example:
```suggestion
    } catch (Throwable t) {
      // delete ColumnView to avoid memory leak
      deleteColumnView(viewHandle);
      viewHandle = 0;
      throw t;
    }
```